test/replication-luatest/gh_9720_tmp_spaces_ddl_abort_applier_txs_test.lua

```diff
@@ -0,0 +1,166 @@
+    cg.replica_set = replica_set:new{}
+    local box_cfg = {
+        memtx_use_mvcc_engine = cg.params.memtx_use_mvcc,
+        replication = {
+            server.build_listen_uri('server1', cg.replica_set.id),
+            server.build_listen_uri('server2', cg.replica_set.id),
+        },
+        replication_timeout = 0.1,
+    }
+    cg.server1 = cg.replica_set:build_and_add_server{
+        alias = 'server1',
+        box_cfg = box_cfg,
+    }
+    box_cfg.read_only = true
+    cg.server2 = cg.replica_set:build_and_add_server{
+        alias = 'server2',
+        box_cfg = box_cfg,
+    }
+    cg.replica_set:start()
+    cg.server1:exec(function()
+        local s = box.schema.create_space('s')
+        s:create_index('p')
+        box.begin()
+        for i = 1, 1000 do
+            s:replace{i, i}
+        end
+        box.commit()
+    end)
+    cg.server2:exec(function()
+        box.cfg{read_only = false}
+    end)
+    cg.replica_set:wait_for_fullmesh()
+    cg.server1:wait_for_downstream_to(cg.server2)
+end)
+
+g.after_each(function(cg)
+    cg.replica_set:drop()
+end)
+
+-- Test that a DDL transaction on fully temporary spaces does not abort a remote
+-- transaction.
+g.test_fully_temporary_ddl_does_not_abort_remote_tx = function(cg)
+    t.tarantool.skip_if_not_debug()
+
+    cg.server2:exec(function()
+        box.error.injection.set('ERRINJ_CHECK_FORMAT_DELAY', true)
+    end)
+    cg.server1:exec(function()
+        box.space.s:format{'i', 'unsigned'}
+    end)
+    cg.server2:exec(function()
+        box.schema.space.create('tmp', {type = 'temporary'})
+        box.error.injection.set('ERRINJ_CHECK_FORMAT_DELAY', false)
+    end)
+    cg.server1:wait_for_downstream_to(cg.server2)
+    cg.server2:exec(function()
+        local msg = "Transaction has been aborted by conflict"
+        t.assert_not_equals(box.info.replication[1].upstream.message, msg)
+        t.assert_not_equals(box.space.s:format(), {})
+    end)
+end
+
+-- Test that a DDL transaction on fully temporary spaces still aborts a not pure
+-- remote transaction.
+g.test_fully_temporary_ddl_aborts_not_pure_remote_tx = function(cg)
+    t.skip_if(not cg.params.memtx_use_mvcc)
+
+    cg.server2:exec(function()
+        local tmp = box.schema.space.create('tmp-replace', {type = 'temporary'})
+        tmp:create_index('p')
+
+        box.space.s:on_replace(function()
+            tmp:replace{0}
+            rawset(_G, 'executed_on_replace', true)
+            while true do require('fiber').yield() end
+        end)
+    end)
+    cg.server1:exec(function()
+        box.space.s:replace{0}
+    end)
+    cg.server2:exec(function()
+        t.helpers.retrying({timeout = 60}, function()
+            t.assert(_G.executed_on_replace)
+        end)
+        box.schema.space.create('tmp', {type = 'temporary'})
+    end)
+    local msg = "Transaction committing DDL (id=%d+) has aborted " ..
+                "another TX (id=%d+)"
+    cg.server2:grep_log(msg)
```

Comment:
`grep_log` won't fail on its own, you need to check it with `t.assert`.

Reply:
Thanks, fixed!